coderd/notifications/manager.go

```
@@ -161,11 +160,6 @@ func (m *Manager) loop(ctx context.Context) error {
 	}()
 
 	m.mu.Lock()
-	if m.closed {
-		m.mu.Unlock()
-		return xerrors.New("manager already closed")
```

I personally don't think we need to guard against improper API usage, where `Run` is explicitly called after `Stop`, but I'm not completely opposed to the idea of dropping an error log if that happens.

---

This change may cause `Stop` to grab the mutex first, checks `m.notifier = nil` and unlocks. Then this code-path locks, assigns `m.notifier` and executes `m.notifier.run`. The `m.notifier.stop` is never called.

Can't we just enable the slog ignore errors option for the failing test, or change the `m.log.Error` to warning/debug instead?